### PR TITLE
[Feature] Support apply_token_bitmask_inplace_cpu for Bf16/Fp16.

### PR DESCRIPTION
**Pull request by Seven-Streams**
As requested in #447, this PR supports Bf16/Fp16 logits for `apply_token_bitmask_inplace_cpu`. With this PR, `apply_token_bitmask_inplace_cpu` can support `float32`, `bfloat16`, and `float16` logits.

### CODE REVIEW

File: cpp/grammar_matcher.cc [hunk 1]
```diff
@@ -85,6 +86,73 @@ std::pair<bool, int> _IsSingleTokenBitmask(const DLTensor& bitmask, int vocab_si
   }
 }
 
+void ApplyMaskFP32(
```
Comment (Ubospica) [inline cpp/grammar_matcher.cc]: Fp32

**Review by Ubospica — COMMENTED** [review]
Could you also add triton kernels?

**Comment by Seven-Streams** [comment]
> Could you also add triton kernels?

Current `cuda`, `triton`, `torch_compile` has supported it.
